hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -67,6 +67,7 @@ object PruneDeadFields {
         case tir: TableIR =>
           memoizeTableIR(tir, tir.typ, memo)
           rebuild(tir, memo)
+        case bmir: BlockMatrixIR => bmir //NOTE Currently no BlockMatrixIRs would have dead fields
```

Comment:
This is fine for now; or you could build the appropriate memo/rebuild that passes over the existing BlockMatrixIR but does nothing.  We will want to add IR nodes that go to/from MatrixTables (see for example, MatrixTable.writeBlockMatrix in Scala and BlockMatrix.entries in Python) and even the block matrices themselves don't get pruned, the nested (Matrix)Table IR inside of them should be.